HW04.Rmd

```diff
@@ -1,126 +1,215 @@
+
+## Read in the data
+
+The gun_deaths.csv is provided in the repository. 
+```{r load-data, message=FALSE, warning=FALSE}
+library(tidyverse)    # load tidyverse packages, including ggplot2
+library(readr)
+library(knitr)
+library(usethis)
+
+# read in the data
+gun_deaths <- read_csv("gun_deaths.csv")
+```
+
+## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
+
+Knitr is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
+
+
+```{r month, warning=FALSE}
+death_month <- gun_deaths %>% 
+  group_by(month) %>% 
+  summarise('Number of Gun Deaths' = n())
+kable(death_month)
+```
+
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+p1 <- ggplot(death_month, aes(x = month.abb[month], y = `Number of Gun Deaths`)) +
+  geom_bar(stat = "identity", color = "skyblue", fill = "skyblue") +
+  labs(title = 'Number of Gun Deaths per Month', x = "Month") +
+  theme(plot.title = element_text(hjust = 0.5))
+
+p1
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+death_intent <- gun_deaths %>% 
+  group_by(intent) %>% 
+  summarise('Number of Gun Deaths' = n())
+
+p2 <- ggplot(death_intent, aes(x = reorder(intent, -`Number of Gun Deaths`), y = `Number of Gun Deaths`)) +
+  geom_bar(stat = 'identity', color = 'rosybrown', fill = 'rosybrown') +
+  labs(title = 'Number of Gun Deaths by Intent', x = "Month") +
+  theme(plot.title = element_text(hjust = 0.5))
+
+p2
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+# Generate a boxplot visualizing the age of gun death victims, by sex
+gun_deaths$age <- as.numeric(gun_deaths$age)
+
+p3 <- ggplot(gun_deaths, aes(x = sex, y = age)) +
+  geom_boxplot()
+
+p3
+
+# Print the averae age of female gun death victims
+mean(gun_deaths %>% filter(sex == "F") %>% select(age) %>% unlist(), na.rm = TRUE)
+```
+
+
+## How many white males with at least a high school education were killed by guns in 2012?
+
+```{r education}
+gun_deaths %>% 
+  filter(year == 2012, sex == "M", race == "White", education %in% c('HS/GED', 'BA+', 'Some college')) %>% 
+  summarise(n())
+```
+
+`Answer:` 15199 white males with at least a high school education were killed by guns in 2012. 
+
+
+## Which season of the year has the most gun deaths?
+
+Assume that:
+
+* Winter = January-March
+* Spring = April-June
+* Summer = July-September
+* Fall = October-December
+
+```{r season}
+gun_season <- gun_deaths %>% 
+  mutate(season = ifelse(month %in% 1:3, "Winter",
```

Comment:
The way you use ifelse() to generate season variable is very concise!

Reply:
That's a useful function to learn!